### PR TITLE
[ALLUXIO-2090] New UFS integration tests for pagination

### PR DESCRIPTION
**Pull request by madanadit**
Tests if list and delete work correctly for directories large enough to require pagination.

https://alluxio.atlassian.net/browse/ALLUXIO-2090

### CODE REVIEW

File: tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java [hunk 1]
```diff
@@ -195,6 +238,46 @@ public void listTest() throws IOException {
...
+    // Make top level directory
+    mUfs.mkdirs(topLevelDirectory, false);
+
+    // Make the children files
+    for (int i = 0; i < numFiles; ++i) {
+      createEmptyFile(PathUtils.concatPath(topLevelDirectory, filePrefix
+          + String.format("%04d", i)));
+    }
+    // Make the children folders
+    for (int i = 0; i < numFolders; ++i) {
```
Comment (yupeng9) [inline tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java]: These two tests share quite some amount of code. Shall we extract them into a `prepare` method?

Comment (madanadit) [inline tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java]: Absolutely. @yupeng9 Thanks for the suggestion!.

**Comment by madanadit** [comment]
@yupeng9 Please take another look when you get a chance. Thanks.

**Comment by yupeng9** [comment]
LGTM. Thanks for adding this test.

**Pull request merged**